source/adapters/level_zero/queue.cpp

```diff
@@ -1313,10 +1312,22 @@ ur_result_t urQueueReleaseInternal(ur_queue_handle_t Queue) {
       for (auto &QueueGroup : QueueMap)
         for (auto &ZeQueue : QueueGroup.second.ZeQueues)
           if (ZeQueue) {
-            auto ZeResult = ZE_CALL_NOCHECK(zeCommandQueueDestroy, (ZeQueue));
-            // Gracefully handle the case that L0 was already unloaded.
-            if (ZeResult && ZeResult != ZE_RESULT_ERROR_UNINITIALIZED)
-              return ze2urResult(ZeResult);
+            if (UrL0QueueSyncNonBlocking) {
```

Comment:
we can only get here if the Queue's refcount reaches 0. Which shouldn't happen while there's a synchronize happening on that queue in parallel.

Reply:
I will recheck to see if there is any possibility of the queue to be deleted while a thread is calling the sync, if not, then I will remove the mutex entirely.